packages/controls/src/controls/shape/v2/shape-v2.test.ts

```diff
@@ -34,388 +30,152 @@ function deserializer(record: DeserializedRecord): ControlDefinition {
+        label: 'Shape popover',
+        layout: 'makeswift::controls::shape-v2::layout::popover',
         type: {
-          val: Color({ defaultValue: 'red' }),
-          num: Checkbox(),
+          background: {
+            config: {
+              label: 'Background color',
+            },
+            type: 'makeswift::controls::color',
+            version: 1,
+          },
+          shape: {
+            type: 'makeswift::controls::shape-v2',
+            config: {
+              label: 'Inline shape',
+              layout: 'makeswift::controls::shape-v2::layout::inline',
+              type: {
+                checkbox: {
+                  config: {
+                    label: 'Checkbox',
+                  },
+                  type: 'makeswift::controls::checkbox',
+                  version: 1,
+                },
+                color: {
+                  config: {
+                    label: 'Color',
+                    defaultValue: 'red',
+                  },
+                  type: 'makeswift::controls::color',
+                  version: 1,
+                },
+              },
+            },
+          },
         },
-      })
-      expect(shape).toMatchSnapshot()
-    })
-
-    test('disallows extraneous properties', () => {
-      ShapeV2({
-        type: {
-          toggle: Checkbox(),
+      },
+    }
+
+    const group = new GroupDefinition(
+      {
+        label: 'Shape popover',
+        preferredLayout: Group.Layout.Popover,
+        props: {
+          background: Color({ label: 'Background color' }),
+          shape: new GroupDefinition(
+            {
+              label: 'Inline shape',
+              preferredLayout: Group.Layout.Inline,
+              props: {
+                checkbox: Checkbox({
+                  label: 'Checkbox',
+                }),
+                color: Color({ label: 'Color', defaultValue: 'red' }),
+              },
+            },
+            ShapeV2Definition.v1DataType,
+          ),
         },
-        // @ts-expect-error
-        extra: 'extra',
-      })
-    })
-  })
-
-  describe('assignability', () => {
-    function assignTest(_def: ShapeV2Definition) {}
-    assignTest(
-      ShapeV2({
-        type: {},
-        label: 'Shape',
-      }),
-    )
-
-    assignTest(
-      ShapeV2({
-        type: {},
-      }),
+      },
+      ShapeV2Definition.v1DataType,
     )
 
-    assignTest(
-      ShapeV2({
-        type: {
-          toggle: Checkbox(),
-          color: Color({ defaultValue: 'red' }),
-        },
-      }),
+    const deserialized = ShapeV2Definition.deserialize(
+      deserializeRecord(serializedRecord as any as SerializedRecord),
+      deserializer,
     )
 
-    assignTest(
-      ShapeV2({
-        type: {
-          combo: Combobox({
-            getOptions: () => [{ id: '0', value: 'a', label: 'Alpha' }],
-          }),
-          link: Link({ label: 'Link' }),
-        },
-      }),
-    )
+    expect(deserialized).toStrictEqual(group)
```

Comment:
Test that we correctly deserialize `ShapeV2` serialized record into a corresponding `Group` with the `ShapeV2` data tag.